discovery/gossiper.go

```diff
@@ -1361,6 +1299,68 @@ func (d *AuthenticatedGossiper) networkHandler() {
 	}
 }
 
+// handleNetworkMessages is responsible for waiting for dependencies for a
+// given network message and processing the message. Once processed, it will
+// signal its dependants and add the new announcements to the announce batch.
+//
+// NOTE: must be run as a goroutine.
+func (d *AuthenticatedGossiper) handleNetworkMessages(nMsg *networkMsg,
+	deDuped *deDupedAnnouncements, vb *routing.ValidationBarrier,
+	shouldBroadcast bool) {
+
+	go func() {
```

Comment:
We're already invoking this with `go handleNetworkMessages` so no need to spin up another goroutine.

Reply:
oops yeah my bad, now fixed!